Common.mk

```diff
@@ -457,36 +456,44 @@ DEPENDENCY_TOOLS=buildctl helm jq lz4 skopeo tuftool yq
 # around checksum mismatch
 FORCE_BUILD_ON_HOST?=false
 
-# $1 - target(s)
-# $2 - should run on host
+# $1 - should run on host
 MAYBE_RUN_IN_DOCKER?=$(if \
 	$(or \
 		$(filter true,$(IS_ON_BUILDER_BASE)), \
 		$(filter true,$(FORCE_BUILD_ON_HOST)), \
 		$(filter true,$2), \
-		$(filter false,$(EVAL_DOCKER_AVAILABLE)) \
-	),,$(foreach target,$1,run-in-docker/$(target)))
+		$(filter false,$(DOCKER_AVAILABLE)) \
+	),,true)
 
 # this can be used as a normal macro, $(ENABLE_DOCKER), or as a func with 1 param, $(call ENABLE_DOCKER)
 # $1 - should run on host (optional)
-ENABLE_DOCKER=$(ENABLE_LOGGING)$(eval $(call ENABLE_DOCKER_BODY,$@,$(if $(filter undefined,$(origin 1)),false,$(value 1)),))$(DOCKER_TARGET)
+ENABLE_DOCKER=$(eval $(call ENABLE_DOCKER_BODY,$@,$(if $(filter undefined,$(origin 1)),false,$(value 1)),))$(if $(USE_DOCKER),ensure-docker,)
 # $1 - container platform to use
-ENABLE_DOCKER_PLATFORM=$(ENABLE_LOGGING)$(eval $(call ENABLE_DOCKER_BODY,$@,false,$(if $(filter undefined,$(origin 1)),,$(value 1))))$(DOCKER_TARGET)
+ENABLE_DOCKER_PLATFORM=$(eval $(call ENABLE_DOCKER_BODY,$@,false,$(if $(filter undefined,$(origin 1)),,$(value 1))))$(if $(USE_DOCKER),ensure-docker,)
 
 # $1 - target name
 # $2 - should run on host
-# SHELL is overriden to the cli `true` so that when the actual target ends up running on the host it is an noop
+# $3 - container platform to use (for cgo)
+# SHELL is overriden with the run_in_docker.sh $(RUN_IN_DOCKER_BODY) args passed at the beginning
+# in the make_shell.sh, if the action is docker, then it will wait until its the actual target
+# signified by LOGGING_TARGET being set and then run that script instead of the actual target script
 # since we are manipulating the SHELL here we have to make sure any variables which are going to be used in the
-# context of MAYBE_RUN_IN_DOCKER cannot themselves call out to a shell, we force eval to avoid
+# context of MAYBE_RUN_IN_DOCKER cannot themselves call out to a shell, we can force eval to avoid
 define ENABLE_DOCKER_BODY
 $(eval _TARGET:=$1)
 $(eval _RUN_ON_HOST:=$2)
 $(eval _DOCKER_PLATFORM:=$3)
-$(_TARGET): EVAL_DOCKER_AVAILABLE:=$$(DOCKER_AVAILABLE)
-$(_TARGET): EVAL_RUN_ON_HOST:=$(_RUN_ON_HOST)
-$(_TARGET): DOCKER_PLATFORM:=$(_DOCKER_PLATFORM)
-$(_TARGET): DOCKER_TARGET=$$(call MAYBE_RUN_IN_DOCKER,$$@,$$(EVAL_RUN_ON_HOST))
-$(_TARGET): SHELL=$$(if $$(DOCKER_TARGET),$$(NOOP_SHELL),$$(LOGGING_SHELL))
+$(eval _USE_DOCKER:=$(call MAYBE_RUN_IN_DOCKER,$(_RUN_ON_HOST)))
+$(_TARGET): export LOGGING_TARGET=$(_TARGET)
+$(_TARGET): export RUN_IN_DOCKER_ARGS=$$(RUN_IN_DOCKER_ARGS_BODY)
+$(_TARGET): MAKE_TARGET=$(_TARGET)
+$(_TARGET): DOCKER_PLATFORM=$(_DOCKER_PLATFORM)
+$(_TARGET): USE_DOCKER=$(_USE_DOCKER)
+$(_TARGET): SHELL=$(if $(_USE_DOCKER),$(DOCKER_SHELL),$(LOGGING_SHELL))
+endef
+
+define RUN_IN_DOCKER_ARGS_BODY
```

Comment:
broke this out to support using it in the actual target, in case someone wants to run it manually, and to export it for the implicit target.